psutil/arch/windows/process_info.c

```diff
@@ -162,6 +162,72 @@ const int STATUS_INFO_LENGTH_MISMATCH = 0xC0000004;
 const int STATUS_BUFFER_TOO_SMALL = 0xC0000023L;
 
 
+
+#define WINDOWS_UNITIALIZED 0
+#define WINDOWS_XP 51
+#define WINDOWS_VISTA 60
+#define WINDOWS_7 61
+#define WINDOWS_8 62
+#define WINDOWS_81 63
+#define WINDOWS_10 100
+
+
+int get_windows_version() {
+    OSVERSIONINFO ver_info;
+    BOOL result;
+    DWORD dwMajorVersion;
+    DWORD dwMinorVersion;
+    DWORD dwBuildNumber;
+    static int windows_version = WINDOWS_UNITIALIZED;
+    // didn't get version yet
+    if (windows_version == WINDOWS_UNITIALIZED) {
+        memset(&ver_info, 0, sizeof(ver_info));
+        ver_info.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+        result = GetVersionEx(&ver_info);
+        if (result != FALSE) {
+            dwMajorVersion = ver_info.dwMajorVersion;
+            dwMinorVersion = ver_info.dwMinorVersion;
+            dwBuildNumber = ver_info.dwBuildNumber;
+            // Windows XP, Windows server 2003
+            if (dwMajorVersion == 5 && dwMinorVersion == 1) {
+                windows_version = WINDOWS_XP;
+            }
+            // Windows Vista
+            else if (dwMajorVersion == 6 && dwMinorVersion == 0) {
+                windows_version = WINDOWS_VISTA;
+            }
+            // Windows 7, Windows Server 2008 R2
+            else if (dwMajorVersion == 6 && dwMinorVersion == 1) {
+                windows_version = WINDOWS_7;
+            }
+            // Windows 8, Windows Server 2012
+            else if (dwMajorVersion == 6 && dwMinorVersion == 2) {
+                windows_version = WINDOWS_8;
+            }
+            // Windows 8.1, Windows Server 2012 R2
+            else if (dwMajorVersion == 6 && dwMinorVersion == 3)
+            {
+                windows_version = WINDOWS_81;
+            }
+            // Windows 10, Windows Server 2016
+            else if (dwMajorVersion == 10) {
+                windows_version = WINDOWS_10;
+            }
+        }
+    }
+    return windows_version;
+}
+
+_NtQueryInformationProcess get_NtQueryInformationProcess() {
+    static _NtQueryInformationProcess NtQueryInformationProcess = NULL;
+    if (NtQueryInformationProcess == NULL) {
+        NtQueryInformationProcess = (_NtQueryInformationProcess)GetProcAddress(
+            GetModuleHandleA("ntdll.dll"), "NtQueryInformationProcess");
+    }
+    return NtQueryInformationProcess;
+}
```

Comment:
I like putting this in a utility function because it can be reused. Would you mind using this utility function also in other parts of the code? Also, for consistency I think it's better to rename this function to `psutil_NtQueryInformationProcess`.